### PR TITLE
ATB-1377 - Feature Tests: Added Merge Feature-tests to main workflow

### PR DESCRIPTION
**Pull request by ChrisBates1**
updated name for publish feature tests to dev workflow added merge feature-tests to main workflow

### What changed

Updated the Publish feature-tests to Dev workflow
Added the Publish feature tests to main Workflow (Build ECR Test Image Push)

### Why did it change

To enable test images to be pushed to the Build ECR Repo and feature-tests to be run on the AIS Main Pipeline

### Issue tracking

- [ATB-1377](https://govukverify.atlassian.net/browse/ATB-1377)

## Checklists
- [x] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [x] Tested changes and included test evidence in the PR, if appropriate
- [x] Included all required tags and other properties for any new resources in the SAM template
- [x] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [x] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [x] Ensured that no log lines include PII or other sensitive data
- [x] Implemented unit testing for any new logic implemented, if appropriate
- [x] Ensured that all commits in this PR are signed
- [x] Ensured appropriate code coverage is maintained by unit tests
- [x] Checked SonarCube and ensured no code smells were added

[ATB-1377]: https://govukverify.atlassian.net/browse/ATB-1377?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/merge-feature-tests-to-main.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+name: "Publish Feature Tests to Build"
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "feature-tests/**"
+      - ".github/workflows/manual-publish-feature-tests-to-dev.yaml"
```
Comment (domantas-gds) [inline .github/workflows/merge-feature-tests-to-main.yaml:10]:
```suggestion
      - ".github/workflows/merge-feature-tests-to-main.yaml"
```

File: .github/workflows/merge-feature-tests-to-main.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  workflow_dispatch:
+    inputs:
```
Comment (domantas-gds) [inline .github/workflows/merge-feature-tests-to-main.yaml:12]: inputs section can be removed as should not be able to input feature branch into build workflow

File: .github/workflows/merge-feature-tests-to-main.yaml (hunk 1 above)
Comment (ChrisBates1) [reply .github/workflows/merge-feature-tests-to-main.yaml]: Thanks Dom, changed made on latest push

File: .github/workflows/merge-feature-tests-to-main.yaml (hunk 2 above)
Comment (ChrisBates1) [reply .github/workflows/merge-feature-tests-to-main.yaml]: Thanks Dom, changed made on latest push

File: .github/workflows/merge-feature-tests-to-main.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        type: string
+        default: main
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  publish_artifacts:
+    name: "Publish Feature Tests Image to Dev"
```
[the remark this answers is not in the document]
Comment (ChrisBates1) [reply .github/workflows/merge-feature-tests-to-main.yaml]: Thanks Dom, changed made on latest push

**Review by MinaMahdi — APPROVED** [review]

**Review by liamquayle — APPROVED** [review]

**Pull request merged**